src/qibo/abstractions/circuit.py

```diff
@@ -912,3 +912,97 @@ def read_args(args):
                 gate_list[i] = ("M", qubit_list, register)
 
         return len(qubits), gate_list
+
+    def draw(self, line_wrap=None) -> str: # pragma: no cover
+        """Draw text circuit using unicode symbols.
+
+        Args:
+            line_wrap (int): maximum number of characters per line. This option
+                split the circuit text diagram in chunks of line_wrap characters.
+
+        Return:
+            String containing text circuit diagram.
+        """
+        labels = {"h": "H", "x": "X", "y": "Y", "z": "Z",
+                  "rx": "RX", "ry": "RY", "rz": "RZ",
+                  "u1": "U1", "u2": "U2", "u3": "U3",
+                  "cx": "NOT", "swap": "x", "cz": "Z",
+                  "crx": "RX", "cry": "RY", "crz": "RZ",
+                  "cu1": "U1", "cu3": "U3", "ccx": "X",
+                  "id": "I", "collapse": "M", "fsim": "f",
+                  "generalizedfsim": "gf"}
+
+        # build string representation of gates
+        matrix = [[] for _ in range(self.nqubits)]
+        idx = [0] * self.nqubits
+
+        for gate in self.queue:
+            gate_name = labels.get(gate.name)
```

Comment:
There are some gates such as the `CallbackGate`, `Flatten` and the channels which are not included in `labels`. In this case using `labels.get` here will return a `None` which will raise some errors later which may be hard for the user to understand. I would suggest to include a check like:
```Python
if gate.name not in labels:
    raise_error(NotImplementedError, f"{gate.name} gate is not supported by `circuit.draw`")
```